sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Helpers.cs

```
@@ -118,7 +118,8 @@ internal static bool IsJson(string input)
             }
 
             // try parsing input to json object
-            var jsonObj = JToken.Parse(input);
+            var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(input));
```

Couple of options

1. Use `JsonDocument.Parse(input);` with a try catch block around it.
example: 
```
try
{
    var parsed = JsonDocument.Parse(input);
    return parsed != null;
}
catch { 
    return false; 
}
```

2. Check the the return value of `TryParseValue`. it doesn't throw by default but does return true or false if the parsing succussed. it's a small check. 
example: 
```
if (JsonDocument.TryParseValue(ref reader, out JsonDocument jsonObj))
{
    return jsonObj != null;
}
return false;
            ```

---

also use `using` with JsonDocument. It's `IDisposable`

---

We would want to throw an exception from this method because when we catch the exception where it is called from and throw  RequestValidationException or ResponseValidationException accordingly

---

Cool, then use Parse instead of TryParseValue

---

Yup, done that, thanks